src/userspace/exec_monitor.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef EXEC_MONITOR_H
+#define EXEC_MONITOR_H
+
+#include <vector>
+#include <string>
+
+class ExecMonitor
+{
+	std::vector<int> ppid_list;
+	std::vector<std::string> name_list;
+	int n_proc; /* number of tasks to record */
+
+public:
+	ExecMonitor();
+	ExecMonitor(std::vector<int> ppid_list, std::vector<std::string> name_list, int n_proc);
```

Comment:
Why have two constructors when you'll only ever call one?
I'd recommend having only one constructor that takes a reference to a Config object and keeps it, this will save you from having to copy everything and making this constructor giant.